conformance/google-java/src/test/kotlin/com/connectrpc/conformance/Conformance.kt

```diff
@@ -49,36 +49,38 @@ import org.junit.runner.RunWith
 import org.junit.runners.Parameterized
 import org.junit.runners.Parameterized.Parameters
 import org.testcontainers.containers.GenericContainer
+import org.testcontainers.containers.wait.strategy.HostPortWaitStrategy
 import java.time.Duration
 import java.util.Base64
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 
-
 @RunWith(Parameterized::class)
 class Conformance(
-    private val protocol: NetworkProtocol
+    private val clientProtocol: NetworkProtocol,
+    private val serverProtocol: NetworkProtocol
```

Comment:
I'm a little confused as to why we need a server and client protocol which differ considering they need to be using the same network protocol. From the looks of the code below, it seems like this is only being used to determine whether to run the server using `connect-go` or `grpc-go`. Would it make sense to run all of these conformance tests for:
- `connect-go`: gRPC, gRPC-Web, Connect protocols
- `grpc-go`: gRPC protocol

For context, this is what Connect-Swift does: https://github.com/connectrpc/connect-swift/blob/fad1776a89fb9e9089ca728c6e55f9fe58f48968/Tests/ConnectLibraryTests/ConnectConformance/ConformanceConfiguration.swift#L46-L59

Reply:
Yeah was just re-using an existing enum here. This expands from testing:
* client: gRPC, server: Connect
* client: Connect, server: Connect

to also test:
* client: gRPC, server: gRPC

There is follow on work to support gRPC-web and I'd also like to get parameterized tests that run on HTTP/1.1 (for connect) as well.

Reply:
I think it's worth renaming them to indicate that you're trying to hit a `connect-go` or `grpc-go` service because with the existing framing it's convoluting protocols and libraries

Reply:
Introduced a new ServerType enum.